micro-rdk/src/common/ota.rs

```diff
@@ -0,0 +1,359 @@
+                log::error!("connection failed: {:?}", err);
+            }
+        });
+        let request = Request::builder()
+            .method("GET")
+            .uri(uri)
+            .body(Empty::<Bytes>::new())
+            .map_err(|e| OtaError::Other(e.to_string()))?;
+        let mut response = sender
+            .send_request(request)
+            .await
+            .map_err(|e| OtaError::Other(e.to_string()))?;
+
+        if response.status() != 200 {
+            return Err(OtaError::Other(
+                format!("Bad Request - Status:{}", response.status()).to_string(),
+            ));
+        }
+        let headers = response.headers();
+        log::debug!("headers: {:?}", headers);
+
+        if !headers.contains_key(hyper::header::CONTENT_LENGTH) {
+            log::error!("header");
+            return Err(OtaError::Other(
+                "response header missing content length".to_string(),
+            ));
+        }
+        let file_len = headers[hyper::header::CONTENT_LENGTH]
+            .to_str()
+            .expect("failed to get reference to content length")
+            .parse::<usize>()
+            .map_err(|e| OtaError::Other(e.to_string()))?;
+
+        if file_len > OTA_MAX_IMAGE_SIZE {
+            return Err(OtaError::InvalidImageSize(file_len));
+        }
+
+        #[cfg(feature = "esp32")]
+        let (mut ota, running_fw_info) = {
+            let ota = EspOta::new().map_err(OtaError::EspError)?;
+            let fw_info = ota.get_running_slot().map_err(OtaError::EspError)?.firmware;
+            (ota, fw_info)
+        };
+        #[cfg(feature = "esp32")]
+        let mut update_handle = ota.initiate_update().map_err(|_| OtaError::InitError)?;
+        #[cfg(not(feature = "esp32"))]
+        let mut update_handle = Vec::new();
+        let mut nwritten: usize = 0;
+        let mut total_downloaded: usize = 0;
+        let mut got_info = false;
+
+        while let Some(next) = response.frame().await {
+            let frame = next.unwrap();
+            if !frame.is_data() {
+                return Err(OtaError::Other(
+                    "download contained non-data frame".to_string(),
+                ));
+            }
+            let data = frame.into_data().unwrap();
+            total_downloaded += data.len();
+
+            if !got_info {
+                if total_downloaded < core::mem::size_of::<EspAppDesc>() {
+                    log::error!("initial frame too small to retrieve esp_app_desc_t");
+                } else {
+                    log::info!("data length {}", data.len());
+                    #[cfg(feature = "esp32")]
+                    {
+                        let mut loader = EspFirmwareInfoLoader::new();
+                        loader.load(&data).map_err(OtaError::EspError)?;
+                        let new_fw = loader.get_info().map_err(OtaError::EspError)?;
+                        log::info!("current firmware: {:?}", running_fw_info);
```

Comment:
`log::debug` instead (for both statements?

Reply:
will do

Reply:
Overall though, I do feel like this method should probably have some well thought out user facing logs that provide context about what the OTA system is doing.